src/odemis/gui/cont/correlation.py

```diff
@@ -0,0 +1,242 @@
+
+# -*- coding: utf-8 -*-
+
+"""
+@author: Rinze de Laat, Éric Piel, Philip Winkler, Victoria Mavrikopoulou,
+         Anders Muskens, Bassim Lazem
+
+Copyright © 2012-2022 Rinze de Laat, Éric Piel, Delmic
+
+Handles the switch of the content of the main GUI tabs.
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License version 2 as published by the Free
+Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+
+"""
+import logging
+import numpy
+import wx
+# IMPORTANT: wx.html needs to be imported for the HTMLWindow defined in the XRC
+# file to be correctly identified. See: http://trac.wxwidgets.org/ticket/3626
+# This is not related to any particular wxPython version and is most likely permanent.
+import wx.html
+from odemis import model
+import odemis.acq.stream as acqstream
+import odemis.gui.model as guimod
+from odemis.acq.stream import StaticStream
+
+# TODO: move to more approprate location
+def updateImageInViews(s: StaticStream, views: list):
+    """force update the static stream in the selected views
+    param: s (StaticStream) the static stream to update
+    param: views (list[View]) the list of views to update"""
+    v: guimod.View 
+    for v in views:
+        for sp in v.stream_tree.getProjections():  # stream or projection
+            if isinstance(sp, acqstream.DataProjection):
+                st = sp.stream
+            else:
+                st = sp
+            if st is s:
+                sp._shouldUpdateImage()
+
+class CorrelationController(object):
+
+    def __init__(self, tab_data, panel, tab, viewports):
+        """
+        tab_data (MicroscopyGUIData): the representation of the microscope GUI
+        panel (wx._windows.Panel): the panel containing the UI controls
+        tab: (Tab): the tab which should show the data
+        viewports: (ViewPorts): the tab view ports
+        """
+        
+        self._tab_data_model = tab_data
+        self._main_data_model = tab_data.main
+        self._panel = panel
+        self._tab = tab
+        self._viewports = viewports
+
+        # the correlation streams, and selected stream
+        self.correlationStreams = model.ListVA()
+        self.selected_stream = model.VigilantAttribute(None)
```

Comment:
That represent the state of the GUI, so it should go on the model: CryoCorrelationGUIData.
Note that all tab models have a `.streams` VA which should be enough to use, instead of `correlationStreams`. 
For `selected_stream`, I'd recommend the casing `selectedStream`, to match the rest of that file.